### PR TITLE
Support field in dataclass member initialization

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards #2276

### CODE REVIEW

File: integration_tests/structs_35.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from lpython import dataclass, field, i32
+from numpy import array
+
+@dataclass
+class X:
+    a: i32 = 123
+    b: bool = True
+    c: list[i32] = field(default_factory=lambda: [1, 2, 3])
+    d: i32[3] = field(default_factory=lambda: array([4, 5, 6]))
+    e: i32 = field(default=-5)
```
Comment (certik) [inline integration_tests/structs_35.py:10]: Is this the same as:
```suggestion
    e: i32 = -5
```
?

**Review by certik — APPROVED** [review]
I think this looks good.

File: integration_tests/structs_35.py (hunk 1 above)
Comment (Shaikh-Ubaid) [reply integration_tests/structs_35.py:10]: Yes, I think they are equivalent.

Comment (certik) [reply integration_tests/structs_35.py:10]: If they are equivalent, then I would almost not support the duplicate way of doing it ("just one way of doing things").

**Pull request merged**